src/irmin-pack/dict.ml

```diff
@@ -78,11 +80,12 @@ module Make (IO : IO.S) : S = struct
     let log_offset = IO.force_offset t.io in
     if log_offset > former_log_offset then refill ~from:former_log_offset t
 
+  let ro_sync t = if IO.readonly t.io then sync_offset t
```

Comment:
out if curiosity, does it matter if we try to sync a RW instance? Why not allowing this?

Reply:
The memory and disk versions of the `log` and `log_async` in a RW instance are always in sync (modulo flushing), so there is nothing to do in that case.

Reply:
so can we just call that function `sync`? (and that would be a no-op for RW instances)

Reply:
I think it might be better to raise an exception in the RW case; it's probably indicative of a misunderstanding of the API. Perhaps an argument in favour of phantom capabilities (or a completely different functor instantiation).

Reply:
It now raises an exception, as in index.